src/fs.ts

```diff
@@ -127,24 +129,42 @@ export class FileSystemUpdater {
 	 * This function cannot be cancelled because multiple callers get the result of the same operation.
 	 *
 	 * @param uri URI of the file to ensure
+	 * @param childOf A parent span for tracing
 	 */
-	ensure(uri: string): Promise<void> {
-		return this.fetches.get(uri) || this.fetch(uri);
+	async ensure(uri: string, childOf = new Span()): Promise<void> {
+		const span = childOf.tracer().startSpan('Ensure file content', { childOf });
+		span.addTags({ uri, fresh: !this.fetches.has(uri) });
+		try {
+			return await (this.fetches.get(uri) || this.fetch(uri));
+		} catch (err) {
+			span.setTag('error', true);
```

Comment:
yeah, thought about that too, but couldn't settle on a good API. Would leave it as this for now.